daemons/based/based_callbacks.c

```diff
@@ -528,8 +528,11 @@ parse_local_options_v2(const pcmk__client_t *cib_client, int call_type,
                        gboolean *process, gboolean *needs_forward)
 {
     if (cib_op_modifies(call_type)) {
-        if (pcmk__str_any_of(op, PCMK__CIB_REQUEST_PRIMARY,
-                             PCMK__CIB_REQUEST_SECONDARY, NULL)) {
+        if (pcmk_is_set(call_options, cib_scope_local)
```

Comment:
Working on getting the early commits broken out into a separate PR. I'm running CTSlab. It turns out this one causes a whole bunch of
```
May 16 12:21:49	BadNews: May 16 12:21:19 fastvm-rhel-9-0-42 pacemaker-based[73860]:  notice: Local CIB 0.27.244.e377a0bc5b6774df5df515e07f2953fa differs from fastvm-rhel-9-0-44: 0.27.240.90650fca63694fdb350e07d30d7b38d9 0x562b2b354dc0
May 16 12:21:49	BadNews: May 16 12:21:19 fastvm-rhel-9-0-42 pacemaker-based[73860]:  notice: Local CIB 0.27.244.e377a0bc5b6774df5df515e07f2953fa differs from fastvm-rhel-9-0-45: 0.27.240.90650fca63694fdb350e07d30d7b38d9 0x562b2b974bf0
May 16 12:21:49	BadNews: May 16 12:21:19 fastvm-rhel-9-0-42 pacemaker-based[73860]:  notice: Local CIB 0.27.244.e377a0bc5b6774df5df515e07f2953fa differs from fastvm-rhel-9-0-46: 0.27.240.90650fca63694fdb350e07d30d7b38d9 0x562b2b98bd00
May 16 12:21:49	BadNews: May 16 12:21:19 fastvm-rhel-9-0-42 pacemaker-based[73860]:  notice: Local CIB 0.27.244.e377a0bc5b6774df5df515e07f2953fa differs from fastvm-rhel-9-0-43: 0.27.240.90650fca63694fdb350e07d30d7b38d9 0x562b2b95c020
```

(not just the occasional known-issue ones)

I'll have to look into it. Surely `cib_scope_local` should make a CIB change local-only... Maybe `cib_scope_local` is being misused somewhere, since we've never done this in the past.

Reply:
Okay, I think the best bet is to deprecate `cib_scope_local` rather than trying to "fix" or extend it. It was nothing but a hack for legacy mode, which forwarded messages with `host == NULL` unless `cib_scope_local` was set or the local node was primary.

A few modifying requests use it for various reasons, and those would have to be handled if we extended `cib_scope_local` for non-legacy mode. Deprecating it is easier.

Reply:
Addressed in #3105, where I moved a lot of the preliminary stuff to de-clutter this PR